server/registry/storage_test.go

```diff
@@ -0,0 +1,75 @@
+// Copyright 2022 Google LLC. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package registry
+
+import (
+	"context"
+	"testing"
+
+	"github.com/apigee/registry/rpc"
+	"github.com/apigee/registry/server/registry/internal/storage/models"
+	"github.com/apigee/registry/server/registry/names"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
+)
+
+func TestDuplicateCreation(t *testing.T) {
```

Comment:
I added this to assure myself that `AlreadyExists` was being returned when I expected - so in one sense, it's development scaffolding, but in another I think it's useful to verify an underlying assumption. I see this as a test that particularly defends against changes in the gorm dependency. If the code returned by gorm changes, it would be easier to detect and diagnose here than by stepping through the API method tests. 

I had hoped to put this inside the `storage` package, but the database configuration that it needed was in the `registry` (server) package. We might rearrange that in the future if we decide to add specific storage package tests.

Reply:
Mentioned [here](https://github.com/apigee/registry/pull/640/files#r915198299)